DNN Platform/Library/Services/Log/EventLog/LogProperties.cs

```diff
@@ -25,8 +29,54 @@ public string Summary
 
                 return summary;
             }
-        }
-
+        }
+
+        /// <inheritdoc />
+        ILogDetailInfo IList<ILogDetailInfo>.this[int index]
+        {
+            get => (ILogDetailInfo)this[index];
+            set => this[index] = value;
+        }
+
+        /// <inheritdoc />
+        public int IndexOf(ILogDetailInfo item) =>
+            base.IndexOf(item);
+
+        /// <inheritdoc />
+        public void Insert(int index, ILogDetailInfo item) =>
+            base.Insert(index, item);
+
+        /// <inheritdoc />
+        public void Add(ILogDetailInfo item) => 
+            base.Add(item);
+
+        /// <inheritdoc />
+        public bool Contains(ILogDetailInfo item) => 
+            base.Contains(item);
+
+        /// <inheritdoc />
+        public void CopyTo(ILogDetailInfo[] array, int arrayIndex) => 
+            base.CopyTo(array, arrayIndex);
+
+        /// <inheritdoc />
+        IEnumerator<ILogDetailInfo> IEnumerable<ILogDetailInfo>.GetEnumerator() => 
+            this.ToArray().Cast<ILogDetailInfo>().GetEnumerator();
+
+        /// <inheritdoc />
+        public bool Remove(ILogDetailInfo item)
+        {
+            try
+            {
+                base.Remove(item);
+            }
+            catch
+            {
+                return false;
+            }
+
+            return true;
+        }

```

Comment:
@bdukes do you have any thoughts on this implementation. The interface requires a bool return yet `ArrayList` is just void. I wrapped it to catch any exception and return false.

Open to ideas here.

Reply:
I think you'd probably need to do a contains check first.  Unless the `ArrayList` is going to throw if the item doesn't exist.

Reply:
I just looked at the ArrayList implementation for .NET 5 which may differ slightly, but it is a useful reference. It doesn't appear to throw any exceptions with this method. If it can't find the item it just slightly continues.

https://github.com/dotnet/runtime/blob/de6f9114a54587ea1c3e13c1aba2dbaff3c39fc5/src/libraries/System.Private.CoreLib/src/System/Collections/ArrayList.cs#L536-L557
```c#
        public virtual void Remove(object? obj)
        {
            int index = IndexOf(obj);
            if (index >= 0)
                RemoveAt(index);
        }


        // Removes the element at the given index. The size of the list is
        // decreased by one.
        //
        public virtual void RemoveAt(int index)
        {
            if (index < 0 || index >= _size) throw new ArgumentOutOfRangeException(nameof(index), SR.ArgumentOutOfRange_Index);


            _size--;
            if (index < _size)
            {
                Array.Copy(_items, index + 1, _items, index, _size - index);
            }
            _items[_size] = null;
            _version++;
        }
```

## Next Steps
I agree with you, let's update this to check if the item exists and then delete it